Unable to upgrade from 0.24-stable to 0.25-stable (npm dependency resolution failing)

### Describe the bug

`decidim:upgrade` fails to work for upgrading from `0.24-stable` to `0.25-stable` (or later):

```bash
decidim@decidim:~/decidim$ RAILS_ENV=production bin/rails decidim:upgrade
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @webpack-cli/configtest@1.1.0
npm WARN Found: webpack-cli@3.3.12
npm WARN node_modules/webpack-cli
npm WARN   webpack-cli@"^3.3.12" from the root project
npm WARN   1 more (@rails/webpacker)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer webpack-cli@"4.x.x" from @webpack-cli/configtest@1.1.0
npm WARN node_modules/@webpack-cli/configtest
npm WARN   @webpack-cli/configtest@"^1.1.0" from webpack-cli@4.9.1
npm WARN   node_modules/@codegram/decidim-bulletin_board/node_modules/webpack-cli
npm WARN 
npm WARN Conflicting peer dependency: webpack-cli@4.10.0
npm WARN node_modules/webpack-cli
npm WARN   peer webpack-cli@"4.x.x" from @webpack-cli/configtest@1.1.0
npm WARN   node_modules/@webpack-cli/configtest
npm WARN     @webpack-cli/configtest@"^1.1.0" from webpack-cli@4.9.1
npm WARN     node_modules/@codegram/decidim-bulletin_board/node_modules/webpack-cli
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @webpack-cli/info@1.4.0
npm WARN Found: webpack-cli@3.3.12
npm WARN node_modules/webpack-cli
npm WARN   webpack-cli@"^3.3.12" from the root project
npm WARN   1 more (@rails/webpacker)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer webpack-cli@"4.x.x" from @webpack-cli/info@1.4.0
npm WARN node_modules/@webpack-cli/info
npm WARN   @webpack-cli/info@"^1.4.0" from webpack-cli@4.9.1
npm WARN   node_modules/@codegram/decidim-bulletin_board/node_modules/webpack-cli
npm WARN 
npm WARN Conflicting peer dependency: webpack-cli@4.10.0
npm WARN node_modules/webpack-cli
npm WARN   peer webpack-cli@"4.x.x" from @webpack-cli/info@1.4.0
npm WARN   node_modules/@webpack-cli/info
npm WARN     @webpack-cli/info@"^1.4.0" from webpack-cli@4.9.1
npm WARN     node_modules/@codegram/decidim-bulletin_board/node_modules/webpack-cli
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @webpack-cli/serve@1.6.0
npm ERR! Found: webpack-cli@3.3.12
npm ERR! node_modules/webpack-cli
npm ERR!   webpack-cli@"^3.3.12" from the root project
npm ERR!   webpack-cli@"^3.3.12" from @rails/webpacker@5.4.3
npm ERR!   node_modules/@rails/webpacker
npm ERR!     @rails/webpacker@"5.4.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer webpack-cli@"4.x.x" from @webpack-cli/serve@1.6.0
npm ERR! node_modules/@webpack-cli/serve
npm ERR!   @webpack-cli/serve@"^1.6.0" from webpack-cli@4.9.1
npm ERR!   node_modules/@codegram/decidim-bulletin_board/node_modules/webpack-cli
npm ERR!     webpack-cli@"^4.2.0" from @codegram/decidim-bulletin_board@0.21.2
npm ERR!     node_modules/@codegram/decidim-bulletin_board
npm ERR!       @codegram/decidim-bulletin_board@"0.21.2" from @decidim/elections@0.25.2
npm ERR!       packages/elections
npm ERR!   dev @webpack-cli/serve@"^1.6.0" from the root project
npm ERR!   1 more (@decidim/webpacker)
npm ERR! 
npm ERR! Conflicting peer dependency: webpack-cli@4.10.0
npm ERR! node_modules/webpack-cli
npm ERR!   peer webpack-cli@"4.x.x" from @webpack-cli/serve@1.6.0
npm ERR!   node_modules/@webpack-cli/serve
npm ERR!     @webpack-cli/serve@"^1.6.0" from webpack-cli@4.9.1
npm ERR!     node_modules/@codegram/decidim-bulletin_board/node_modules/webpack-cli
npm ERR!       webpack-cli@"^4.2.0" from @codegram/decidim-bulletin_board@0.21.2
npm ERR!       node_modules/@codegram/decidim-bulletin_board
npm ERR!         @codegram/decidim-bulletin_board@"0.21.2" from @decidim/elections@0.25.2
npm ERR!         packages/elections
npm ERR!     dev @webpack-cli/serve@"^1.6.0" from the root project
npm ERR!     1 more (@decidim/webpacker)
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/decidim/.npm/_logs/2023-03-05T15_58_50_578Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/decidim/.npm/_logs/2023-03-05T15_58_50_578Z-debug-0.log

== Command npm i --save-prod ./packages/browserslist-config ./packages/core ./packages/elections ./packages/webpacker failed ==
```

### To Reproduce

1. Change from  'release/0.24-stable` to `release/0.25-stable` in your Gemfile.
2. Run `bundle update'
3. Run `bin/rails decidim:upgrade`

### Expected behavior

Upgrades should work.

### Screenshots

_No response_

### Stacktrace

_No response_

### Extra data

- Device: Server
- Device OS: Debian 11
- Browser: NA
- Decidim Version: 0.24 -> 0.25
- Decidim installation: Own


### Additional context

This is my umpteenth attempt upgrading this 0.24 installation. I have never been able to get anywhere close to succeeding.

--- a/docs/modules/install/pages/manual.adoc
+++ b/docs/modules/install/pages/manual.adoc
@@ -12,14 +12,7 @@ In order to develop on decidim, you will need:
 
 The compatibility between the different versions of the components is the following:
 
-|===
-|Decidim version |Ruby version |Node version
-
-|v0.28 | 3.1.1 | 18.17.x
-
-|v0.27 | 3.0.2 | 16.18.x
-
-|===
+include::install:partial$version_matrix.adoc[]
 
 We are starting with an Ubuntu 22.04.3 LTS. This is an opinionated guide, so you are free to use the technology that you are most comfortable. If you have any doubts and you are blocked you can go and ask on https://matrix.to/#/#decidimdevs:matrix.org[our Matrix.org chat room for developers].
 

--- a/docs/modules/install/pages/manual.adoc
+++ b/docs/modules/install/pages/manual.adoc
@@ -10,6 +10,17 @@ In order to develop on decidim, you will need:
 * *ImageMagick*
 * *Chrome* browser and https://sites.google.com/a/chromium.org/chromedriver/[chromedriver].
 
+The compatibility between the different versions of the components is the following:
+
+|===
+|Decidim version |Ruby version |Node version
+
+|v0.28 | 3.1.1 | 18.17.x
+
+|v0.27 | 3.0.2 | 16.18.x
+
+|===
+
 We are starting with an Ubuntu 22.04.3 LTS. This is an opinionated guide, so you are free to use the technology that you are most comfortable. If you have any doubts and you are blocked you can go and ask on https://matrix.to/#/#decidimdevs:matrix.org[our Matrix.org chat room for developers].
 
 We recommend to have at least some basic proficiency in GNU/Linux (i.e. how to use the command-line, packages, etc), networking knowledge, server administration, development in general, and some basic knowledge about software package managers. It would be great to have Ruby on Rails development basics (a good starting point is http://guides.rubyonrails.org/getting_started.html[Getting Started with Ruby on Rails]) and have some knowledge on how package libraries are working (we use `bundler` for handling ruby packages, and `npm`/`yarn` for handling javascript).

--- a/docs/modules/install/pages/update.adoc
+++ b/docs/modules/install/pages/update.adoc
@@ -80,6 +80,11 @@ gem "decidim-initiatives", DECIDIM_VERSION
 gem "decidim-dev", DECIDIM_VERSION
 ----
 
+== Compatiblity versions matrix
+
+There are different versions of Decidim that are compatible with different versions of Ruby and Node. You can check the following table to see which versions are compatible with each other:
+
+include::install:partial$version_matrix.adoc[]
 
 [discrete]
 == Recommendations

--- a/docs/modules/install/partials/version_matrix.adoc
+++ b/docs/modules/install/partials/version_matrix.adoc
@@ -1,0 +1,9 @@
+
+|===
+|Decidim version |Ruby version |Node version
+
+|v0.28 | 3.1.1 | 18.17.x
+
+|v0.27 | 3.0.2 | 16.18.x
+
+|===

--- a/docs/modules/install/partials/version_matrix.adoc
+++ b/docs/modules/install/partials/version_matrix.adoc
@@ -1,9 +1,13 @@
 
 |===
-|Decidim version |Ruby version |Node version
+|Decidim version |Ruby version |Node version | Status
 
-|v0.28 | 3.1.1 | 18.17.x
+|develop | 3.2.2 | 18.17.x | Unreleased
 
-|v0.27 | 3.0.2 | 16.18.x
+|v0.28 | 3.1.1 | 18.17.x | Bug fixes and security updates
+
+|v0.27 | 3.0.2 | 16.18.x | Bug fixes and security updates
+
+|v0.26 | 2.7.5+ | 16.9.x | Not maintained
 
 |===

